Python/sampleclient/filechecks.py

```diff
@@ -0,0 +1,40 @@
+import os
+from pathlib import Path
+
+
+def file_exists(file_path):
+    '''Returns if file exists or not.'''
+    file = Path(file_path)
+    return file.exists()
+
+
+def file_size_in_bytes(file_path):
+    '''Return size of file in bytes, or -1 if file does not exist.'''
+    if not file_exists(file_path):
+        return -1
+    statinfo = os.stat(file_path)
+    return int(statinfo.st_size)
+
+
+MAX_FILE_SIZE_FOR_PUT_API = int(1024 * 1024 * 4)
+
+
+def files_must_not_be_more_than(files, limit):
```

Comment:
I think the DSL-like direction with this is interesting, but I think it is maybe too clever for what this larger script is all about. I feel like we're coming up with new error handling idioms, which makes the overall code harder to understand.

Can we try to make this more "boring"? Can we implement the logic in fewer functions, or maybe with a judiciously chosen exception to catch the case where one of the files doesn't exist?

Reply:
I was thinking about that, but from my perspective I felt this was also a teaching tool of sorts on how to handle the API, so I felt it was best to be more verbose and explicit than we needed to be.

The idea is that customers will be looking at this script to learn how to set up their own automation; We don't want them creating a support request unless they have to so I wanted to effectively give them a tutorial as well.